src/world/message_handler.h

```diff
@@ -23,18 +23,56 @@ along with this program.  If not, see http://www.gnu.org/licenses/
 
 #include "common/mmo.h"
 
+#include <task_system.hpp>
+#ifdef WIN32
+#include <winsock2.h>
+#else
+#include <netinet/in.h>
+#endif
+
+/**
+ * Class responsible for handling messages recieved from the message server.
+ *
+ * Implementations of this class are responsible for queueing any work
+ * that affects internal state / sql queries via the given task system, using
+ * the submit() method.
+ *
+ * This is done to ensure thread-safety outside of the message_server thread.
+ */
 class IMessageHandler
 {
 public:
+    IMessageHandler()
+    {
+        ts = new ts::task_system(1);
```

Comment:
`std::make_unique`

Reply:
But as I mentioned above, I'd look into making this class: `class IMessageHandler : public ts::task_system`

Reply:
You could also make it inherit from `SqlConnection` 👀